driver-core/src/main/java/com/datastax/driver/core/JdkSSLOptions.java

```diff
@@ -74,6 +96,25 @@ protected SSLEngine newSSLEngine(SocketChannel channel) {
         return engine;
     }
 
+    /**
+     * Creates an SSL engine each time a connection is established.
+     * <p/>
+     * <p/>
+     * You might want to override this if you need to fine-tune the engine's configuration
+     * (for example enabling hostname verification).
+     *
+     * @param channel the Netty channel for that connection.
+     * @param address remote address for the connection
+     * @return the engine.
+     */
+    protected SSLEngine newSSLEngine(SocketChannel channel, InetSocketAddress address) {
+        SSLEngine engine = context.createSSLEngine(address.getHostName(), address.getPort());
```

Comment:
It might be a better idea to make `newSSLEngine(SocketChannel channel)` delegate to `newSSLEngine(SocketChannel channel, InetSocketAddress address)` and test the nullity of parameter `address`. This would reduce the amount of duplicated code.